pallets/asset/src/identifier.rs

```diff
@@ -0,0 +1,143 @@
+use codec::{Decode, Encode};
+use std::thread::sleep_ms;
+
+#[derive(Encode, Decode, Clone, Debug, PartialEq, Eq)]
+pub enum Identifier {
```

Comment:
Doc comments on all of the variants here would make things substantially less cryptic for the uninitiated. :)